docs/workflow/testing/coreclr/testing.md

```diff
@@ -1,43 +1,300 @@
+
+On Windows:
+
+```cmd
+.\src\tests\run.cmd x64 checked
+```
+
+On MacOS/Linux:
+
+```bash
+./src/tests/run.sh -x64 -checked
+```
+
+The `run.cmd/run.sh` scripts also have a number of flags you can pass to set specific conditions and environment variables, such as _JIT Stress_, _GC Stress_, and so on. Run it with only any one of the help flags for more details.
+
+Once your tests are done running, a report will be generated with all the results under the `artifacts/log` folder, and will be named `TestRun_<Arch>_<Configuration>.html`. The tests that failed will be listed in a file called `TestRunResults_<OS>_<Arch>_<Configuration>.err`.
+
+For individual test results and outputs, those are written in a `Reports` folder within the test root's directory (`artifacts/tests/coreclr/<OS>.<Arch>.<Configuration>`). For example, the results for the _JIT Intrinsics Math Round Double_ test shown in [Building an Individual Test](#building-an-individual-test), would be placed in `artifacts/tests/coreclr/<OS>.<Arch>.<Configuration>/Reports/JIT/Intrinsics/MathRoundDouble_ro`.
+
+### Running Individual Tests
+
+After you've built one (or more) tests, the way to run them is by calling the generated `cmd/sh` script alongside them. These scripts take three optional arguments:
+
+* `-debug`: Receives the path of a debugger to run the test under in.
+* `-env`: Path to a _.env_ file to specify environment variables to be set for the test. More info about _dotenv_ can be found in [their repo](https://github.com/motdotla/dotenv).
+* -coreroot: The path to the Core_Root you wish to use. Note that this flag is mandatory unless you have the `CORE_ROOT` environment variable set. Then, you can omit it and the script will use that one.
+
+If this list of parameters feels familiar, it's because it's virtually the same as the arguments that `corerun` receives. More info on `corerun` in its [how-to-use doc](/docs/workflow/testing/using-corerun-and-coreroot.md).
+
+These scripts have a couple more hidden functionalities, which can be activated by setting their environment variables prior to running the script:
+
+* Run with Crossgen2: `RunCrossGen2=1`
+* Build and run as composite: `CompositeBuildMode=1`. Note that this one depends on `RunCrossGen2` being set.
+
+Let's run one of the Intrinsics tests as an example:
+
+On Windows:
+
+```cmd
+set CORE_ROOT=<repo_root>\artifacts\tests\coreclr\windows.<Arch>.<Configuration>\Tests\Core_Root
+cd path\to\JIT\Intrinsics\MathRoundDouble_ro
+.\MathRoundDouble_ro.cmd
+```
+
+On MacOS/Linux:
+
+```bash
+export CORE_ROOT=<repo_root>/artifacts/tests/coreclr/<OS>.<Arch>.<Configuration>/Tests/Core_Root
```

Comment:
You can also pass `coreroot=<repo_root>/artifacts/tests/coreclr/<OS>.<Arch>.<Configuration>/Tests/Core_Root` argument to the tests .sh. This is my preferred way. Similarly on Windows, but this time the `coreroot` is separated by space from the path, not by `=`.